relay-filter/src/config.rs

```diff
@@ -192,6 +196,83 @@ impl LegacyBrowsersFilterConfig {
     }
 }
 
+/// Configuration for a generic filter.
+#[derive(Clone, Debug, Default, Serialize, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct GenericFilterConfig {
+    /// Specifies whether this filter is enabled.
+    pub is_enabled: bool,
+    /// The condition for the filter.
+    pub condition: RuleCondition,
+}
+
+impl GenericFilterConfig {
+    /// Returns true if the filter is not enabled or the condition is not supported.
+    pub fn is_empty(&self) -> bool {
+        !self.is_enabled || matches!(self.condition, RuleCondition::Unsupported)
+    }
+}
+
+/// Configuration for generic filters.
+#[derive(Clone, Debug, Default)]
+pub struct GenericFiltersConfig(pub Vec<(String, GenericFilterConfig)>);
```

Comment:
Yep, exactly. It's for matching purposes, since this list is now dynamic and i want deterministic behaviors with respect to the order defined in the JSON.